catalogs/catalog-fileset/src/main/java/org/apache/gravitino/catalog/fileset/FilesetCatalogOperations.java

```diff
@@ -257,6 +240,44 @@ public void initialize(
       this.defaultFileSystemProvider =
           FileSystemUtils.getFileSystemProviderByName(
               fileSystemProvidersMap, defaultFileSystemProviderName);
+
+      scheduler =
+          new ScheduledThreadPoolExecutor(
+              1,
+              new ThreadFactoryBuilder()
+                  .setDaemon(true)
+                  .setNameFormat("file-system-cache-for-fileset" + "-%d")
+                  .build());
+
+      Caffeine<Object, Object> cacheBuilder =
+          Caffeine.newBuilder()
+              .expireAfterAccess(1, TimeUnit.HOURS)
+              .removalListener(
+                  (ignored, value, cause) -> {
+                    try {
+                      ((FileSystem) value).close();
+                    } catch (IOException e) {
+                      LOG.warn("Failed to close FileSystem instance in cache", e);
+                    }
+                  })
+              .scheduler(Scheduler.forScheduledExecutorService(scheduler));
+
+      MetricsSystem metricsSystem = GravitinoEnv.getInstance().metricsSystem();
+      // Metrics System could be null in UT.
+      if (metricsSystem != null) {
+        cacheBuilder.recordStats(
+            () -> {
+              String metricsPrefix =
+                  String.join(
+                      ".",
+                      GRAVITINO_FILESET_CATALOG_METRIC_PREFIX,
+                      catalogInfo.namespace().toString(),
+                      catalogInfo.name(),
+                      FILESYSTEM_CACHE);
+              return new MetricsStatsCounter(metricsSystem.getMetricRegistry(), metricsPrefix);
```

Comment:
seems you should unregister the metrics when closing fileset catalog

Reply:
Does it seem unnecessary? The metrics source is not being used here.